File: machine/cpp/machine/memory/map.cpp
#include "machine/memory/map.h"
#include "spec/bmsx/memory_map.h"

namespace bmsx {

uint32_t RAM_SIZE = BMSX_RAM_BYTES;
uint32_t RAM_END = RAM_BASE + RAM_SIZE;

void configureMemoryMap(uint32_t ramBytes) {
	RAM_SIZE = ramBytes;
	RAM_END = RAM_BASE + RAM_SIZE;
}

} // namespace bmsx

File: machine/ts/machine/memory/map.ts
import { BMSX_RAM_BYTES, RAM_BASE } from '../../spec/bmsx/memory_map';

export let RAM_SIZE = BMSX_RAM_BYTES;
export let RAM_END = RAM_BASE + RAM_SIZE;

export function configureMemoryMap(ramBytes: number): void {
	RAM_SIZE = ramBytes;
	RAM_END = RAM_BASE + RAM_SIZE;
}
